src/NUnitFramework/tests/Attributes/RepeatableTestsWithTimeoutAttributesTests.cs

```diff
@@ -0,0 +1,170 @@
+// ***********************************************************************
+// Copyright (c) 2018 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Commands;
+using NUnit.TestUtilities;
+
+#if THREAD_ABORT
+
+namespace NUnit.Framework.Attributes
+{
+    [TestFixture]
+    public class RepeatableTestsWithTimeoutAttributesTests
+    {
+        private int retryOnlyCount;
+        private int retryTimeoutCount;
+        private int repeatTimeoutCount;
+
+        [Test]
+        [Retry(3)]
+        public void ShouldPassAfter3Retries()
+        {
+            retryOnlyCount++;
+            Assert.True(retryOnlyCount >= 2);
+        }
+
+        [Test]
+        [Retry(3), Timeout(30)]
+        public void ShouldPassAfter3RetiresAndTimeoutIsResetEachTime()
```

Comment:
Fyi, you only got one of the two instances I mentioned of the typo `Retires` (here's the other) 😊